sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_shared/exceptions.py

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+from typing import TYPE_CHECKING
+
+import six
+
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
+
+from ._generated.v2016_10_01.models import KeyVaultErrorException as KeyVaultError2016
+from ._generated.v7_0.models import KeyVaultErrorException as KeyVaultErrorV7
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Union
+
+    KeyVaultErrorExceptionType = Union[KeyVaultError2016, KeyVaultErrorV7]
+
+_error_map = {404: ResourceNotFoundError, 409: ResourceExistsError}
+
+
+def raise_from_key_vault_exception(exception):
+    # type: (KeyVaultErrorExceptionType) -> None
+    """Raises more specific exception types from autorest-generated KeyVaultErrorException"""
+
+    new_class = None
+    try:
+        new_class = _error_map[exception.response.status_code]
+    except (AttributeError, KeyError):
+        # exception has no response/status code, or status code isn't in the error map
+        pass
+
+    if new_class:
+        new_exception = new_class(message=exception.message, response=exception.response)
+        try:
+            new_exception.error = exception.error
+        except AttributeError:
+            pass
+        six.raise_from(new_exception, exception)
+
+    raise exception
```

Comment:
What's the difference between this version and the version in `azure-keyvault-keys`?

Reply:
There is none, `_shared` is copied across Key Vault libraries.

Reply:
I guess that's because the key vault is split in to two separate installable packages? Ouch :/